test/framework/clusterctl/repository.go

```diff
@@ -141,7 +141,7 @@ func CreateRepository(ctx context.Context, input CreateRepositoryInput) string {
 	for key := range input.E2EConfig.Variables {
 		clusterctlConfigFile.Values[key] = input.E2EConfig.GetVariable(key)
 	}
-	clusterctlConfigFile.write()
+	Expect(clusterctlConfigFile.write()).To(Succeed())
```

Comment:
```suggestion
	Expect(clusterctlConfigFile.write()).To(Succeed(), "Failed to write clusterctl config")
```
Let's always add context to expect statements in the framework (same below)